java-connectors/kafka-connect-common/src/main/java/io/lenses/streamreactor/common/util/JarManifest.java

```diff
@@ -69,11 +73,13 @@ public JarManifest(JarFile jarFile) {
     Manifest manifest;
     try {
       Optional<JarFile> jarFileOptional = of(jarFile);
-      manifest = jarFileOptional.get().getManifest();
+      if (jarFileOptional.isPresent()) {
+        manifest = jarFileOptional.get().getManifest();
+        extractMainAttributes(manifest.getMainAttributes());
+      }
     } catch (NullPointerException | IOException e) {
```

Comment:
We shouldn't really need to be catching a NPE if we are already using an Optional?

Reply:
things were a little bit more complicated here as it's `getManifest()` that could give us null. Still, it wasn't neat so I changed that.